libs/stats/odc/stats/_gm_ls_bitmask.py

```diff
@@ -0,0 +1,172 @@
+            |   |   |   |   |   |   |   |   o---------------------------------> water
+            |   |   |   |   |   |   |   x-------------------------------------> cloud_confidence
+            |   |   |   |   |   |   o-----------------------------------------> cloud_confidence
+            |   |   |   |   |   x---------------------------------------------> cloud_shadow_confidence
+            |   |   |   |   o-------------------------------------------------> cloud_shadow_confidence
+            |   |   |   x-----------------------------------------------------> snow_ice_confidence
+            |   |   o---------------------------------------------------------> snow_ice_confidence
+            |   x-------------------------------------------------------------> cirrus_confidence
+            0-----------------------------------------------------------------> cirrus_confidence
+        """
+        mask_band = xx[self.mask_band]
+        xx = xx.drop_vars([self.mask_band])
+
+        # set cloud_mask (dilated_cloud + cloud + cloud_shadow) bitmask - True=cloud, False=non-cloud
+        cloud_mask = da.bitwise_and(mask_band, 0b0000_0000_0001_1010) != 0
+
+        # set no_data bitmask - True=data, False=no-data
+        keeps = da.bitwise_and(mask_band, 0b0000_0000_0000_0001) == 0
+
+        # drops nodata pixels and add cloud_mask from xx
+        xx = keep_good_only(xx, keeps)
+        xx["cloud_mask"] = cloud_mask
+
+        return xx
+
+    def input_data(self, task: Task) -> xr.Dataset:
+
+        chunks = {"y": -1, "x": -1}
+
+        xx = load_with_native_transform(
+            task.datasets,
+            bands=self.bands + [self.mask_band],
+            geobox=task.geobox,
+            native_transform=self._native_tr,
+            fuser=self._fuser,
+            groupby="solar_day",
+            resampling=self.resampling,
+            chunks=chunks,
+        )
+
+        return xx
+
+    def reduce(self, xx: xr.Dataset) -> xr.Dataset:
+        scale = 0.0000275
+        offset = -0.2
+        return_SR = False
+        cfg = dict(
+            maxiters=1000,
+            num_threads=1,
+            scale=scale,
+            offset=offset,
+            return_SR=return_SR,
+            reshape_strategy="mem",
+            out_chunks=(-1, -1, -1),
+            work_chunks=self.work_chunks,
+            compute_count=True,
+            compute_mads=True,
+        )
+
+        cloud_mask = xx["cloud_mask"]
+        xx = xx.drop_vars(["cloud_mask"])
+
+        # erase pixels with cloud
+        xx = erase_bad(xx, cloud_mask)
+
+        gm = geomedian_with_mads(xx, **cfg)
+        gm = gm.rename(self.renames)
+
+        # Scale USGS Landsat bands into surface reflectance
+        if return_SR:
+            for band in gm.data_vars:
+                if band in self.bands:
+                    gm[band] = (scale*10000 * gm[band]+offset*10000).astype(gm[band].dtype)
```

Comment:
@cbur24 `stats` already unscales the `geomedian` and `edev` bands. And `edev` can just be scaled so something like this should work:

```suggestion
                    gm[band] = 10000 * gm[band]
            gm['edev'] = 10000 * gm['edev']
```

Unfortunately though the `bcdev` and `sdev` bands aren't translation invariant and are computed with the scaled data - but that's a seperate issue!

Reply:
@kieranricardo Africa's USGS Landsat Collection 2 data is scaled very differently form Australia's.  `Stats` returns the data in its original scaling, which is not equivalent to surface reflectance for USGS L2C2. This code snippet is to reapply the USGS's scale and offset factors so we can save the data in surface reflectance values (i.e. ARD values) rather than their raw digital numbers (which the user would then have to apply a scale and offset each time they loaded the gm).  
The multiplication by 10,000 is to ensure the Landsat GM matches the same range of values as our Sentinel-2 gm product, and to allow for saving the data in integer format because USDS SR is 0-1.  So in other words, the code should be fine as it is.
 
The `edev` stuff is more complicated because we aren't sure that stats is reapplying the scale and offset factors as it should, so we need to look into it further.

Reply:
@cbur24 makes sense!

If you're still looking for it here's the line of code that transforms `edev` back into it's original scaling:

https://github.com/opendatacube/odc-tools/blob/aaaa4e673016e3a7d5b3c97834412a2bb9635a10/libs/algo/odc/algo/_geomedian.py#L438

so something like this should work:

```
gm['edev'] = 10000 * scale *gm['edev']
```

To avoid bias I think transformed geomedian should be rounded before being caste to an int:

```suggestion
                    gm[band] = np.round((scale*10000 * gm[band]+offset*10000)).astype(gm[band].dtype)
```

And I have no idea if this ends up being an issue or not but scaling the geomedian output by `0.275` (I think thats what `scale * 10_000` ends up being) and casting to an integer ends up decreasing the precision by 4x. I don't what the precision of the input data actually is so sorry if this completely irrelevant!

Reply:
> so something like this should work:
> 
> ```
> gm['edev'] = 10000 * scale *gm['edev']
> ```
>
> To avoid bias I think the transformed geomedian should be rounded before being caste to an int:
>

Both above sound good.
 
> And I have no idea if this ends up being an issue or not but scaling the geomedian output by `0.275` (I think thats what `scale * 10_000` ends up being) and casting to an integer ends up decreasing the precision by 4x. I don't know what the precision of the input data actually is so sorry if this is completely irrelevant!

It's reflectance between 0 - 1 scaled to 0-10000, so rounding error smaller than 0.01% which is more than enough.

Reply:
Added the requested changes but haven't actually tested if the `xarray.DataArray.round()` method works because my sandbox is busy with other processing right now. Will test when I get a chance